### PR TITLE
fix rocm_3.8.0 build error for rocprim,rocrand,rocfft packages

### PR DESCRIPTION
**Pull request by srekolam**

### CODE REVIEW

File: var/spack/repos/builtin/packages/hip/package.py [hunk 1]
```diff
@@ -82,7 +82,10 @@ def get_rocm_prefix_info(self):
                           for name in ('llvm-amdgpu', 'hsa-rocr-dev',
                                        'rocminfo', 'rocm-device-libs'))
             mydict['rocm-path'] = self.spec.prefix
-            device_lib_path = mydict['rocm-device-libs'].amdgcn.bitcode
+            if '@:3.8.0' in self.spec:
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/hip/package.py]: Should this be `self.spec.satisfies('@:3.8.0'):`?

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Refer to use of `self.spec.satisfies` question.

Also please fix the flake8 errors.

File: var/spack/repos/builtin/packages/hip/package.py (hunk 1 above)
Comment (haampie) [reply var/spack/repos/builtin/packages/hip/package.py]: To be fair, the whole file is using `[version] in [spec]`, so for consistency this is better.

**Review by haampie — APPROVED** [review]

**Comment by haampie** [comment]
Approving (modulo flake8 errors)

File: var/spack/repos/builtin/packages/hip/package.py (hunk 1 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/hip/package.py]: > To be fair, the whole file is using `[version] in [spec]`, so for consistency this is better.

I should've looked at more than the change then.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Please fix the flake8 errors, copied below for convenience:

```
var/spack/repos/builtin/packages/hip/package.py:85: [W291] trailing whitespace
15
var/spack/repos/builtin/packages/hip/package.py:86: [E111] indentation is not a multiple of four
16
var/spack/repos/builtin/packages/hip/package.py:86: [E117] over-indented
17
var/spack/repos/builtin/packages/hip/package.py:88: [E111] indentation is not a multiple of four
18
var/spack/repos/builtin/packages/hip/package.py:88: [E117] over-indented
```

**Comment by srekolam** [comment]
i have fixed the flake8 errors. Also i ran the spack install for rocprim, rocrand, rocblas,hipsparse,hibcub etc packages for rocm-3.8.0,rocm-3.9.0,rocm-3.10.0 and see they pass.

**Review by haampie — APPROVED** [review]

**Comment by srekolam** [comment]
@adamjstewart ,can you please let us know if there is anything that is needed for PR to be merged.

**Comment by srekolam** [comment]
@tldahlgren  , can you please review and approve ..Thx

File: var/spack/repos/builtin/packages/hip/package.py [hunk 2]
```diff
@@ -82,7 +82,10 @@ def get_rocm_prefix_info(self):
...
+                device_lib_path = mydict['rocm-device-libs'].lib
+            else:
+                device_lib_path = mydict['rocm-device-libs'].amdgcn.bitcode
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hip/package.py:88]: I don't this this works at all, does it? Shouldn't it be:
```suggestion
            if '@:3.8.0' in self.spec:
                device_lib_path = mydict['rocm-device-libs'].prefix.lib
            else:
                device_lib_path = mydict['rocm-device-libs'].prefix.amdgcn.bitcode
```

Comment (srekolam) [reply var/spack/repos/builtin/packages/hip/package.py:88]: @adamjstewart , the prefix is already appended in the L81 ..mydict = dict((name, self.spec[name].prefix).

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**